src/python_inspector/resolve_cli.py

```diff
@@ -227,8 +227,67 @@ def resolve_dependencies(
         click.secho("Only one of --json or --json-pdt can be used.", err=True)
         ctx.exit(1)
 
+    resolver_api(
+        requirement_files=requirement_files,
+        setup_py_file=setup_py_file,
+        specifiers=specifiers,
+        python_version=python_version,
+        operating_system=operating_system,
+        index_urls=index_urls,
+        json_output=json_output,
+        pdt_output=pdt_output,
+        netrc_file=netrc_file,
+        max_rounds=max_rounds,
+        use_cached_index=use_cached_index,
+        use_pypi_json_api=use_pypi_json_api,
+        verbose=verbose,
+        analyze_setup_py_insecurely=analyze_setup_py_insecurely,
+        ctx=ctx,
+    )
+
+
+def raise_error(message, ctx):
+    """Raise error."""
+    if ctx:
+        click.secho(message, err=True)
+        ctx.exit(1)
+    else:
+        raise ValueError(message)
```

Comment:
When would we want to raise an error? IMHO we should never fail when calling the CLI. We should always run and selectively return error messages if a proper return code on errors, but not a plain Python stack trace. See how this is in SCTK for instance.

Reply:
This will never raise with the CLI, it will only run when there is no ctx